==13757== Memcheck, a memory error detector
==13757== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==13757== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==13757== Command: /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global
==13757== 
==13757== Conditional jump or move depends on uninitialised value(s)
==13757==    at 0x68EE7D: wasmparser::validator::Validator::end (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x68497C: wasmparser::validator::Validator::payload (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x683BF1: wasmparser::validator::Validator::validate_all (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29D0BC: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x268758: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::validate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x580004: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757==    by 0x262361: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==  Uninitialised value was created by a stack allocation
==13757==    at 0x29CFD0: wasmer_compiler::compiler::Compiler::validate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 
==13757== HEAP SUMMARY:
==13757==     in use at exit: 32,173 bytes in 112 blocks
==13757==   total heap usage: 831 allocs, 719 frees, 142,957 bytes allocated
==13757== 
==13757== 6 bytes in 2 blocks are still reachable in loss record 1 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x660A22: wasmer_compiler::translator::environ::ModuleEnvironment::declare_global_import (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65C3B4: wasmer_compiler::translator::sections::parse_import_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x66A1C9: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757== 
==13757== 8 bytes in 1 blocks are still reachable in loss record 2 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x665FE1: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65BCF4: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x669F9F: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757== 
==13757== 8 bytes in 1 blocks are still reachable in loss record 3 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x665FE1: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65BC9F: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x669F9F: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757== 
==13757== 8 bytes in 1 blocks are still reachable in loss record 4 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x665E20: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x660B6C: wasmer_compiler::translator::environ::ModuleEnvironment::declare_global_import (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65C3B4: wasmer_compiler::translator::sections::parse_import_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x66A1C9: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 8 bytes in 1 blocks are still reachable in loss record 5 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E780A: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2A3551: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287B78: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 9 bytes in 2 blocks are still reachable in loss record 6 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x660A97: wasmer_compiler::translator::environ::ModuleEnvironment::declare_global_import (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65C3B4: wasmer_compiler::translator::sections::parse_import_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x66A1C9: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757== 
==13757== 12 bytes in 1 blocks are still reachable in loss record 7 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x6651F0: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65C7E6: wasmer_compiler::translator::sections::parse_function_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x669C84: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757== 
==13757== 16 bytes in 1 blocks are still reachable in loss record 8 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x6C2D78: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x6C75B6: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6CF700: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6C22C5: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58653D: wasmer_engine::trap::error::RuntimeError::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x57FBB0: wasmer::externals::global::Global::set (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26146B: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757==    by 0x262361: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 16 bytes in 1 blocks are indirectly lost in loss record 9 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x266A06: <alloc::boxed::Box<[T]> as core::clone::Clone>::clone (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x283F87: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x5803BD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x585BB2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x260BB9: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757==    by 0x262361: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 24 bytes in 1 blocks are indirectly lost in loss record 10 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x266AF6: <alloc::boxed::Box<[T]> as core::clone::Clone>::clone (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x283F5A: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x5803BD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x585BB2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x260BB9: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757==    by 0x262361: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 26 bytes in 3 blocks are still reachable in loss record 11 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x660D6F: wasmer_compiler::translator::environ::ModuleEnvironment::declare_func_export (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65DA3B: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x66A20D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757== 
==13757== 32 bytes in 1 blocks are still reachable in loss record 12 of 48
==13757==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x4E3F7F4: _dlerror_run (dlerror.c:140)
==13757==    by 0x4E3F165: dlsym (dlsym.c:70)
==13757==    by 0x73F583: fetch (weak.rs:61)
==13757==    by 0x73F583: get<unsafe extern "C" fn(*const libc::unix::linux_like::linux::gnu::b64::x86_64::pthread_attr_t) -> usize> (weak.rs:46)
==13757==    by 0x73F583: min_stack_size (thread.rs:449)
==13757==    by 0x73F583: std::sys::unix::thread::Thread::new (thread.rs:49)
==13757==    by 0x2E08A6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E3C26: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E47DF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 40 bytes in 1 blocks are still reachable in loss record 13 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x6CED5B: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6C2D68: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x6C75B6: backtrace::lock::lock (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6CF700: backtrace::backtrace::trace (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6C22C5: backtrace::capture::Backtrace::new_unresolved (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58653D: wasmer_engine::trap::error::RuntimeError::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x57FBB0: wasmer::externals::global::Global::set (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26146B: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757== 
==13757== 52 bytes in 1 blocks are possibly lost in loss record 14 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x658FC2: hashbrown::raw::inner::RawTable<T>::fallible_with_capacity (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x658471: hashbrown::raw::inner::RawTable<T>::reserve_rehash (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6593DA: hashbrown::raw::inner::RawTable<T>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6698F5: indexmap::map::core::IndexMapCore<K,V>::insert_full (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x657055: indexmap::map::IndexMap<K,V,S>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x660B3E: wasmer_compiler::translator::environ::ModuleEnvironment::declare_global_import (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65C3B4: wasmer_compiler::translator::sections::parse_import_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x66A1C9: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 52 bytes in 1 blocks are possibly lost in loss record 15 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x658FC2: hashbrown::raw::inner::RawTable<T>::fallible_with_capacity (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6589DE: _ZN9hashbrown3raw5inner17RawTable$LT$T$GT$14reserve_rehash17he29be5fc862d84d8E.llvm.8588116468508767673 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65DB71: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x66A20D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 64 bytes in 4 blocks are still reachable in loss record 16 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E9E0C: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2EAC87: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E40B6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 64 bytes in 4 blocks are still reachable in loss record 17 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x73F4DC: alloc (alloc.rs:80)
==13757==    by 0x73F4DC: alloc (alloc.rs:174)
==13757==    by 0x73F4DC: exchange_malloc (alloc.rs:277)
==13757==    by 0x73F4DC: std::sys::unix::thread::Thread::new (thread.rs:44)
==13757==    by 0x2E08A6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E3C26: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E47DF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 72 bytes in 1 blocks are still reachable in loss record 18 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x6466BD: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x64A32B: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x289C35: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x283F15: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x5803BD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x585BB2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x260BB9: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757==    by 0x262361: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 96 bytes in 1 blocks are still reachable in loss record 19 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x6650F7: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65BAA7: wasmer_compiler::translator::sections::parse_type_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x669F9F: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757== 
==13757== 120 bytes in 1 blocks are still reachable in loss record 20 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x664FF7: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65D8C3: wasmer_compiler::translator::sections::parse_export_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x66A20D: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757== 
==13757== 128 bytes in 4 blocks are still reachable in loss record 21 of 48
==13757==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==13757==    by 0x70BB0E: register_dtor (fast_thread_local.rs:33)
==13757==    by 0x70BB0E: try_register_dtor<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:438)
==13757==    by 0x70BB0E: std::thread::local::fast::Key<T>::try_initialize (local.rs:424)
==13757==    by 0x72E309: get<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,fn() -> core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>> (local.rs:410)
==13757==    by 0x72E309: __getit (local.rs:175)
==13757==    by 0x72E309: try_with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:262)
==13757==    by 0x72E309: with<core::cell::RefCell<core::option::Option<std::sys_common::thread_info::ThreadInfo>>,closure-0,()> (local.rs:239)
==13757==    by 0x72E309: std::sys_common::thread_info::set (thread_info.rs:40)
==13757==    by 0x2E0A2C: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x73F8F9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13757==    by 0x73F8F9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13757==    by 0x73F8F9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13757==    by 0x52516DA: start_thread (pthread_create.c:463)
==13757==    by 0x57A271E: clone (clone.S:95)
==13757== 
==13757== 128 bytes in 4 blocks are still reachable in loss record 22 of 48
==13757==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x56C48E1: __cxa_thread_atexit_impl (cxa_thread_atexit_impl.c:106)
==13757==    by 0x2F24DD: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E9FB2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E18B6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E54F9: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E360A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E60A0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E0A6E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x73F8F9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13757==    by 0x73F8F9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13757==    by 0x73F8F9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13757==    by 0x52516DA: start_thread (pthread_create.c:463)
==13757==    by 0x57A271E: clone (clone.S:95)
==13757== 
==13757== 160 bytes in 4 blocks are still reachable in loss record 23 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E7A9C: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E50FB: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2EAEE8: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6030: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E415A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 160 bytes in 4 blocks are still reachable in loss record 24 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E7A9C: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E5126: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2EAEE8: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6030: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E415A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 160 bytes in 4 blocks are still reachable in loss record 25 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E7A9C: std::sync::mutex::Mutex<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2EB0AB: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E41D0: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 160 bytes in 4 blocks are still reachable in loss record 26 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x70CEAE: alloc (alloc.rs:80)
==13757==    by 0x70CEAE: alloc (alloc.rs:174)
==13757==    by 0x70CEAE: exchange_malloc (alloc.rs:277)
==13757==    by 0x70CEAE: new<()> (mutex.rs:223)
==13757==    by 0x70CEAE: std::thread::Thread::new (mod.rs:1149)
==13757==    by 0x2E0791: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E3C26: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E47DF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 160 bytes in 4 blocks are still reachable in loss record 27 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E07B3: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E3C26: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E47DF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 192 bytes in 4 blocks are still reachable in loss record 28 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x729FA6: alloc (alloc.rs:80)
==13757==    by 0x729FA6: alloc (alloc.rs:174)
==13757==    by 0x729FA6: exchange_malloc (alloc.rs:277)
==13757==    by 0x729FA6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13757==    by 0x2E5106: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2EAEE8: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6030: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E415A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 192 bytes in 4 blocks are still reachable in loss record 29 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x729FA6: alloc (alloc.rs:80)
==13757==    by 0x729FA6: alloc (alloc.rs:174)
==13757==    by 0x729FA6: exchange_malloc (alloc.rs:277)
==13757==    by 0x729FA6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13757==    by 0x2E5131: rayon_core::registry::ThreadInfo::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2EAEE8: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6030: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E415A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 192 bytes in 4 blocks are still reachable in loss record 30 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x729FA6: alloc (alloc.rs:80)
==13757==    by 0x729FA6: alloc (alloc.rs:174)
==13757==    by 0x729FA6: exchange_malloc (alloc.rs:277)
==13757==    by 0x729FA6: std::sync::condvar::Condvar::new (condvar.rs:130)
==13757==    by 0x2EB0B1: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E41D0: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 192 bytes in 4 blocks are still reachable in loss record 31 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x70CF12: alloc (alloc.rs:80)
==13757==    by 0x70CF12: alloc (alloc.rs:174)
==13757==    by 0x70CF12: exchange_malloc (alloc.rs:277)
==13757==    by 0x70CF12: new (condvar.rs:130)
==13757==    by 0x70CF12: std::thread::Thread::new (mod.rs:1150)
==13757==    by 0x2E0791: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E3C26: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E47DF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 216 bytes in 1 blocks are still reachable in loss record 32 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x6652E7: alloc::raw_vec::RawVec<T,A>::reserve_exact (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x669919: indexmap::map::core::IndexMapCore<K,V>::insert_full (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x657055: indexmap::map::IndexMap<K,V,S>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x660B3E: wasmer_compiler::translator::environ::ModuleEnvironment::declare_global_import (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x65C3B4: wasmer_compiler::translator::sections::parse_import_section (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x66A1C9: wasmer_compiler::translator::module::translate_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x6601B8: wasmer_compiler::translator::environ::ModuleEnvironment::translate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287694: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 320 bytes in 4 blocks are still reachable in loss record 33 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x70CF61: alloc (alloc.rs:80)
==13757==    by 0x70CF61: alloc (alloc.rs:174)
==13757==    by 0x70CF61: exchange_malloc (alloc.rs:277)
==13757==    by 0x70CF61: new<std::thread::Inner> (sync.rs:324)
==13757==    by 0x70CF61: std::thread::Thread::new (mod.rs:1145)
==13757==    by 0x2E0791: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E3C26: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E47DF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 384 bytes in 1 blocks are still reachable in loss record 34 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E92BC: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E5FEC: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E415A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 416 bytes in 4 blocks are still reachable in loss record 35 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E0848: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E3C26: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E47DF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 504 (464 direct, 40 indirect) bytes in 1 blocks are definitely lost in loss record 36 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x6AE14D: wasmer_vm::instance::InstanceHandle::allocate_instance (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2838E1: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x5803BD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x585BB2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x260BB9: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757==    by 0x262361: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 512 bytes in 1 blocks are still reachable in loss record 37 of 48
==13757==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x72FA19: aligned_malloc (alloc.rs:95)
==13757==    by 0x72FA19: alloc (alloc.rs:22)
==13757==    by 0x72FA19: __rdl_alloc (alloc.rs:304)
==13757==    by 0x2E95D6: alloc::raw_vec::RawVec<T,A>::reserve (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E418A: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 640 bytes in 1 blocks are still reachable in loss record 38 of 48
==13757==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x72FA19: aligned_malloc (alloc.rs:95)
==13757==    by 0x72FA19: alloc (alloc.rs:22)
==13757==    by 0x72FA19: __rdl_alloc (alloc.rs:304)
==13757==    by 0x2E43C6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 640 bytes in 1 blocks are still reachable in loss record 39 of 48
==13757==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x72FA19: aligned_malloc (alloc.rs:95)
==13757==    by 0x72FA19: alloc (alloc.rs:22)
==13757==    by 0x72FA19: __rdl_alloc (alloc.rs:304)
==13757==    by 0x2F28E2: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2F257D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E9FB2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E18B6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E54F9: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E360A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E60A0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E0A6E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 664 bytes in 1 blocks are still reachable in loss record 40 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x287A03: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x26890A: <wasmer_engine_jit::engine::JITEngine as wasmer_engine::engine::Engine>::compile (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x58005F: wasmer::module::Module::from_binary (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25ED11: wasmer::module::Module::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2608D4: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757==    by 0x262361: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 808 bytes in 1 blocks are still reachable in loss record 41 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x651C1F: alloc::collections::btree::map::BTreeMap<K,V>::insert (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x64A731: wasmer_engine::trap::frame_info::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x289C35: <wasmer_engine_jit::artifact::JITArtifact as wasmer_engine::artifact::Artifact>::register_frame_info (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x283F15: wasmer_engine::artifact::Artifact::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x5803BD: wasmer::module::Module::instantiate (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x585BB2: wasmer::instance::Instance::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x260BB9: imported_global::main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x25F1D2: std::rt::lang_start::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x730FB2: {{closure}} (rt.rs:52)
==13757==    by 0x730FB2: do_call<closure-0,i32> (panicking.rs:348)
==13757==    by 0x730FB2: try<i32,closure-0> (panicking.rs:325)
==13757==    by 0x730FB2: catch_unwind<closure-0,i32> (panic.rs:394)
==13757==    by 0x730FB2: std::rt::lang_start_internal (rt.rs:51)
==13757==    by 0x262361: main (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 1,152 bytes in 4 blocks are possibly lost in loss record 42 of 48
==13757==    at 0x4C33B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x4013666: allocate_dtv (dl-tls.c:286)
==13757==    by 0x4013666: _dl_allocate_tls (dl-tls.c:530)
==13757==    by 0x5252227: allocate_stack (allocatestack.c:627)
==13757==    by 0x5252227: pthread_create@@GLIBC_2.2.5 (pthread_create.c:644)
==13757==    by 0x73F62C: std::sys::unix::thread::Thread::new (thread.rs:66)
==13757==    by 0x2E08A6: std::thread::Builder::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E3C26: <rayon_core::registry::DefaultSpawn as rayon_core::registry::ThreadSpawn>::spawn (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E47DF: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 1,520 bytes in 1 blocks are still reachable in loss record 43 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E4222: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2A3551: <wasmer_compiler_cranelift::compiler::CraneliftCompiler as wasmer_compiler::compiler::Compiler>::compile_module (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x287B78: wasmer_engine_jit::artifact::JITArtifact::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 1,536 bytes in 4 blocks are still reachable in loss record 44 of 48
==13757==    at 0x4C33E76: memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x4C33F91: posix_memalign (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x72FA19: aligned_malloc (alloc.rs:95)
==13757==    by 0x72FA19: alloc (alloc.rs:22)
==13757==    by 0x72FA19: __rdl_alloc (alloc.rs:304)
==13757==    by 0x2E9EB0: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2EAC87: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E40B6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 2,008 bytes in 1 blocks are still reachable in loss record 45 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2F202E: crossbeam_epoch::sync::queue::Queue<T>::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2F28B2: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2F257D: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E9FB2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E18B6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E54F9: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E360A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E60A0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E0A6E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x73F8F9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13757==    by 0x73F8F9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13757==    by 0x73F8F9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13757== 
==13757== 4,096 bytes in 4 blocks are still reachable in loss record 46 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E9DF0: crossbeam_deque::deque::Worker<T>::new_lifo (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2EAC87: core::iter::traits::iterator::Iterator::unzip (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E40B6: rayon_core::registry::Registry::new (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E77D6: std::sync::once::Once::call_once::{{closure}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x72A659: std::sync::once::Once::call_inner (once.rs:416)
==13757==    by 0x2E3DA0: rayon_core::registry::global_registry (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E6B9B: rayon_core::current_num_threads (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x28FD81: <rayon::iter::plumbing::bridge::Callback<C> as rayon::iter::plumbing::ProducerCallback<I>>::callback (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x296925: <rayon::iter::while_some::WhileSome<I> as rayon::iter::ParallelIterator>::drive_unindexed (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x29B0D8: rayon::iter::collect::<impl rayon::iter::ParallelExtend<T> for alloc::vec::Vec<T>>::par_extend (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2B6B21: rayon::result::<impl rayon::iter::FromParallelIterator<core::result::Result<T,E>> for core::result::Result<C,E>>::from_par_iter (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757== 
==13757== 6,080 bytes in 4 blocks are still reachable in loss record 47 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2E2FD8: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E60A0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E0A6E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x73F8F9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13757==    by 0x73F8F9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13757==    by 0x73F8F9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13757==    by 0x52516DA: start_thread (pthread_create.c:463)
==13757==    by 0x57A271E: clone (clone.S:95)
==13757== 
==13757== 8,160 bytes in 4 blocks are still reachable in loss record 48 of 48
==13757==    at 0x4C31B0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13757==    by 0x2EF90F: crossbeam_epoch::internal::Local::register (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2F2504: _ZN3std6thread5local4fast12Key$LT$T$GT$14try_initialize17h3d6d30c5335b0a67E.llvm.12191645446024711930 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E9FB2: crossbeam_deque::deque::Stealer<T>::steal (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E18B6: <core::iter::adapters::chain::Chain<A,B> as core::iter::traits::iterator::Iterator>::try_fold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E54F9: rayon_core::registry::WorkerThread::wait_until_cold (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E360A: rayon_core::registry::ThreadBuilder::run (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E60A0: _ZN3std10sys_common9backtrace28__rust_begin_short_backtrace17hd13678e4d803656bE.llvm.9549009445676040077 (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x2E0A6E: core::ops::function::FnOnce::call_once{{vtable-shim}} (in /home/wasmer/repositories/wasmer/wasmer/target/release/examples/imported-global)
==13757==    by 0x73F8F9: call_once<(),FnOnce<()>> (boxed.rs:1081)
==13757==    by 0x73F8F9: call_once<(),alloc::boxed::Box<FnOnce<()>>> (boxed.rs:1081)
==13757==    by 0x73F8F9: std::sys::unix::thread::Thread::new::thread_start (thread.rs:87)
==13757==    by 0x52516DA: start_thread (pthread_create.c:463)
==13757==    by 0x57A271E: clone (clone.S:95)
==13757== 
==13757== LEAK SUMMARY:
==13757==    definitely lost: 464 bytes in 1 blocks
==13757==    indirectly lost: 40 bytes in 2 blocks
==13757==      possibly lost: 1,256 bytes in 6 blocks
==13757==    still reachable: 30,413 bytes in 103 blocks
==13757==         suppressed: 0 bytes in 0 blocks
==13757== 
==13757== For counts of detected and suppressed errors, rerun with: -v
==13757== ERROR SUMMARY: 5 errors from 5 contexts (suppressed: 0 from 0)
